### ISSUE TITLE
Stream Event Parsing Silently Failing and Not Falling Back to Polling

### ISSUE THREAD

**Issue by brandynbennett** [splitio/ruby-client#511]
Streaming events have been failing silently with `JSON::ParserError: Empty input (after ) at line 1, column 1` when trying to parse a streaming event containing a payload missing a leading curly brace. The client did not automatically fallback to polling. The errors were only logged in debug mode and so we weren't alerted to the issue in our production logs. This led to large delays (>30min) from the time a flag definition was updated to the time our application would pick up the change. Only new deploys of the application would pick up new changes.

We are on v8.2.0 of the SDK. 

This is happening [here](https://github.com/splitio/ruby-client/blob/master/lib/splitclient-rb/sse/event_source/event_parser.rb#L46)

Example payload missing leading curly brace:
```json
'data:{"id":"6FMuBkchWr:0:0","timestamp":1710261542842, ... data":"{\"type\":\"SPLIT_UPDATE\", ...
```

**Comment by agustinona** [comment]
Thanks for the report @brandynbennett our team is looking into this. In the meantime, for anyone having this issue it can be resolved by disabling streaming through the factory config: https://help.split.io/hc/en-us/articles/360020673251-Ruby-SDK#configuration

**Comment by agustinona** [comment]
Hi @brandynbennett this issue has been fixed in version 8.3.1

Could you test and confirm?

**Comment by brandynbennett** [comment]
Yes. Just tried this locally and it's behaving as expected. Thank you for the fix.
